### DISCUSSION TITLE
Location based rooms

### DISCUSSION THREAD

**Discussion by scarecrowsowl** [element-hq/element-meta#1341] [Ideas]
This is not an issue, it's a feature request. 
Is it possible to have a feature for location based rooms? Like if I have a local event space, or coffee shop, can it have a room that I only get notifications from when I'm in proximity? I could have a work collaboration that turns off/on when I leave work automatically. This could work great with riot, or you could set a location manually.

**Comment by lampholder** [comment]
This is a neat idea - it might work better with phones + GPS than the web/desktop client with IP/wifi detection, though both would be possible, though you could use low power bluetooth beacons too.

It's not strongly aligned with our current objectives, so I don't think it something the core team is going to progress.

Perhaps most interesting is thinking about what architectural changes we'd need to implement in Riot to make this a feature that could be implemented alongside the more usual usecases. Perhaps some sort of extensible notification management/filter framework could support people filtering notifications except when certain specific criteria are met.

**Comment by almereyda** [comment]
There is a hanging Pull Request vector-im/element-web#2774 which not clearly states what it aims at. Users of What's App have also been reporting, that they can post locations into the stream. Is there anything like a plugin interface to Riot, in so devoted users can manually extend the functionality? Then a location module may provide interesting data and access options. This eventually turns more interesting with events, either in the past, now or in the future. A very basic interface could as minimal as in:

- https://meta.discourse.org/t/locations-plugin/69742
- https://meta.discourse.org/t/events-plugin-calendar/69776

**Comment by t3chguy** [comment]
@almereyda that is not the same thing, this issue is about rooms which become available if you are in a certain location rather than sharing location from within riot.

**Comment by t3chguy** [comment]
@almereyda the PR linked to the one you linked to (https://github.com/matrix-org/matrix-react-sdk/pull/596) is actually for what you want (at least in part)

**Comment by almereyda** [comment]
@t3chguy thanks for the clarification. https://github.com/vector-im/riot-meta/issues/105 also seems related.

I ran into a similar distinction previously in https://github.com/fedwiki/wiki/issues/63#issuecomment-154616274 about *graphs of wiki* and *graphs in wiki*.

**Comment by daniellekirkwood** [comment]
This is a great idea but not something we're currently working on and therefore I'll move this over to our Discussions section!

**Reply by eNkKVS4HxHc9jaq** [reply]
It does sound coo, yet I'm having doubts about the reliability on custom ROMs where in both stock or microG, location services are commonly broken. This'd also consume a fair amount of battery in the background. Overall currently not worth the effort in my opinion.

**Reply by Thatoo** [reply]
I don't think that allowing room owner to tag the location of their own room consume battery in the background.
I don't think that browsing rooms on a map thanks to their location's tag consume more battery than browsing by name.

**Reply by Thatoo** [reply]
Looking for room surrounding us, if room owner set location in their room setting won't consume battery.
But I'm sure it is a feature other chat system don't have and would make Matrix/Element unique.
It would help to reconnect the numerical world to the physical world and enhance user's experience with their own territory.

**Comment by Thatoo** [comment]
Related to https://github.com/vector-im/element-meta/discussions/485 and https://github.com/vector-im/element-meta/discussions/484

**Comment by Thatoo** [comment]
Do we need to write an MSC to ask for adding an optional location settings to rooms?

**Comment by Thatoo** [comment]
And an other MSC to ask for adding an optional location setting in user's settings? 
This location setting could be manually completed by user (or not) or automatically completed (dynamically) by the client (if the user wants it).
